modules/services/window-managers/i3-sway/i3.nix

```diff
@@ -20,8 +26,9 @@ let
         terminal;
 
       keybindings = mkOption {
-        type = types.attrsOf (types.nullOr types.str);
-        default = mapAttrs (n: mkOptionDefault) {
+        type = hm.types.dagOf (types.nullOr types.str);
+        # default = mapAttrs (n: mkOptionDefault) {
```

Comment:
`hm.types.dagOf` doesn't seem compatible with `mkOptionDefault`. Is there some other alternative here :thinking: ?

Reply:
Otherwise this would be a breaking change, and while it doesn't really matter for my use-case (I don't use any of the default keybindings), it is recommended to use `lib.mkOptionDefault` as per documentation.

Reply:
It would be great if you also added support for flags. While they can be passed as-is in the attribute name of the attribute set, they don't go well with overriding the default values because `"$mod+d" != "--no-repeat $mod+d"`.

Reply:
> How does configuring keybindings look like with a dag now?

Dag is mostly the same as a attrset, unless you want to use ordering that than you need to use something like this:https://github.com/nix-community/home-manager/pull/1755/files#diff-43535a9118a6638e49dced945f5385160cbb81112684fea40c4fb196d2a6a02fR71.

> If the UX is worse with dags, I would allow for `types.coercedTo (dagOf ...) (mappingFunction) (attrsOf ...)` so users can keep using the older syntax and convert the attrs notation to a dag transparently

Maybe this would work for supporting `lib.mkOptionDefault`. Supporting it seems important specially because to maintain the correct order for workspaces, we will probably need something like this:

```nix
{
  "${cfg.config.modifier}+1" = hm.dag.entryBefore [ "${cfg.config.modifier}+2" ] "workspace number 1";
  "${cfg.config.modifier}+2" = hm.dag.entryafter [ "${cfg.config.modifier}+1" ] "workspace number 2";
  "${cfg.config.modifier}+3" = hm.dag.entryafter [ "${cfg.config.modifier}+3" ] "workspace number 3";
  "${cfg.config.modifier}+4" = hm.dag.entryafter [ "${cfg.config.modifier}+4" ] "workspace number 4";
  "${cfg.config.modifier}+5" = hm.dag.entryafter [ "${cfg.config.modifier}+5" ] "workspace number 5";
  "${cfg.config.modifier}+6" = hm.dag.entryafter [ "${cfg.config.modifier}+6" ] "workspace number 6";
  "${cfg.config.modifier}+7" = hm.dag.entryafter [ "${cfg.config.modifier}+7" ] "workspace number 7";
  "${cfg.config.modifier}+8" = hm.dag.entryafter [ "${cfg.config.modifier}+8" ] "workspace number 8";
  "${cfg.config.modifier}+9" = hm.dag.entryafter [ "${cfg.config.modifier}+9" ] "workspace number 9";
  "${cfg.config.modifier}+0" = hm.dag.entryafter [ "${cfg.config.modifier}+0" ] "workspace number 10";
}
```

That is well quite ugly to say at least, so it is better to have good defaults and just make people don't need to redefine it.